auth_with_external_browser_test.go

```diff
@@ -0,0 +1,205 @@
+package gosnowflake
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestExternalBrowserSuccessful(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Success, cfg.User, cfg.Password)
+	}()
+	go func() {
+		defer wg.Done()
+		_, err := connectToSnowflake(cfg, "SELECT 1", true)
+		if err != nil {
+			t.Errorf("Connection failed: err %v", err)
+		}
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserFailed(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	cfg.ExternalBrowserTimeout = time.Duration(10000) * time.Millisecond
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Fail, "FakeAccount", "NotARealPassword")
+	}()
+	go func() {
+		defer wg.Done()
+		tOut := "authentication timed out"
+		_, err := connectToSnowflake(cfg, "SELECT 1", false)
+		if err.Error() != tOut {
+			t.Errorf("Expected %v, but got %v", tOut, err)
+		}
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserTimeout(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	cfg.ExternalBrowserTimeout = time.Duration(1000) * time.Millisecond
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Timeout, cfg.User, cfg.Password)
+	}()
+	go func() {
+		defer wg.Done()
+		tOut := "authentication timed out"
+		_, err := connectToSnowflake(cfg, "SELECT 1", false)
+		if err.Error() != tOut {
+			t.Errorf("Expected %v, but got %v", tOut, err)
+		}
+	}()
+	wg.Wait()
+}
+
+func TestExternalBrowserMismatchUser(t *testing.T) {
+	cfg := setupExternalBrowserTest(t)
+	correctUsername := cfg.User
+	cfg.User = "fakeAccount"
+	var wg sync.WaitGroup
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		provideCredentials(externalBrowserType.Success, correctUsername, cfg.Password)
```

Comment:
Why correct username is incorrect?

Reply:
Logically is still correct:
	correctUsername := cfg.User
	cfg.User = "fakeAccount"

but that's my mistake. I will fix it.

Reply:
This TC is testing what's happen if you use correct username to login, BUT in config you provide some wrong user. Basically it triggers specific output:
"390191 (08004): The user you were trying to authenticate as differs from the user currently logged in at the IDP."

Reply:
so the state is ... you are logged as a correct user, but in config you have 'fakeaccount', once trying to execute command it supposed to stops working and return error as above.